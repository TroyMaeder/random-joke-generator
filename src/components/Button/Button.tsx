import React from 'react'
import cn from 'classnames'
import './button.scss'

interface ButtonProps {
    text: string
    className?: string
    onClick?: () => void
}

const Button = ({ text, onClick, className }: ButtonProps) => (
    <button
        className={cn('btn', className)}
        onClick={onClick}
    >
        {text}
    </button>
)

export default Button