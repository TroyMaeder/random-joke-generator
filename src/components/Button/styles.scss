@import '../../shared/constants.scss';

.btn {
    display: flex;
    align-items: center;
    justify-content: center;
}
