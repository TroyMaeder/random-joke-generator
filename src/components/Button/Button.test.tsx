import React from 'react'
import { shallow, configure } from 'enzyme'
import Button from './Button'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('Test Button component', () => {
  it('should be defined', () => {
    expect(Button).toBeDefined()
  })

  it('click event is fired', () => {
    const mockCallBack = jest.fn()

    const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>))
    button.find('button').simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })

  it('should render correctly', () => {
    const tree = shallow(
      <Button text='button text' className="btn" />
    );
    expect(tree).toMatchSnapshot();
  })
})