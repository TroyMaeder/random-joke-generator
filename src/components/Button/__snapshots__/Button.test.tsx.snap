// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component should render correctly 1`] = `ShallowWrapper {}`;
